packages/wrangler/src/type-generation/index.ts

```diff
@@ -118,28 +130,54 @@ export const typesCommand = createCommand({
 		}
 	},
 	async handler(args) {
-		let config: Config;
+		let config: Config | undefined;
 		const secondaryConfigs: Config[] = [];
-		if (Array.isArray(args.config)) {
-			config = readConfig({ ...args, config: args.config[0] });
-			for (const configPath of args.config.slice(1)) {
-				secondaryConfigs.push(readConfig({ config: configPath }));
+		let hasConfigFile = false;
+
+		try {
+			if (Array.isArray(args.config)) {
+				config = readConfig({ ...args, config: args.config[0] });
+				for (const configPath of args.config.slice(1)) {
+					secondaryConfigs.push(readConfig({ config: configPath }));
+				}
+			} else {
+				config = readConfig(args);
 			}
-		} else {
-			config = readConfig(args);
+
+			if (
+				config.configPath &&
+				fs.existsSync(config.configPath) &&
+				!fs.statSync(config.configPath).isDirectory()
+			) {
+				hasConfigFile = true;
+			}
+		} catch (error) {
+			if (args.config) {
+				throw error;
+			}
+			config = undefined;
 		}
 
 		const { envInterface, path: outputPath } = args;
 
-		if (
-			!config.configPath ||
-			!fs.existsSync(config.configPath) ||
-			fs.statSync(config.configPath).isDirectory()
-		) {
-			throw new UserError(
-				`No config file detected${args.config ? ` (at ${args.config})` : ""}. This command requires a Wrangler configuration file.`,
-				{ telemetryMessage: "No config file detected" }
+		if (!hasConfigFile && args.includeRuntime && !args.compatibilityDate) {
+			// Get the maximum compatibility date supported by the installed workerd
+			const miniflareEntry = require.resolve("miniflare");
+			const miniflareRequire = module.createRequire(miniflareEntry);
+			const miniflareWorkerd = miniflareRequire("workerd") as {
+				compatibilityDate: string;
+			};
+			args.compatibilityDate = miniflareWorkerd.compatibilityDate;
+			logger.log(
+				`No config file detected. Using the installed Workers runtime's latest supported date: ${args.compatibilityDate}\n`
+			);
+		}
```

Comment:
There's a utility function for this called `getDevCompatibilityDate()`. Use that instead